// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net.nqe.internal;

// NetworkIDProto contains data that can be used to uniquely identify a network
// type.
// Next id: 4
message NetworkIDProto {
  // Connection type of the network mapped from
  // net::NetworkChangeNotifier::ConnectionType.
  optional int32 connection_type = 1;
  // Name of this network. This is set to WiFi SSID or the MCCMNC of the
  // network.
  optional string id = 2;
  // Signal strength of the network. Set to INT32_MIN when the value is
  // unavailable. Otherwise, must be between 0 and 4 (both inclusive). This may
  // take into account many different radio technology inputs. 0 represents very
  // poor signal strength while 4 represents a very strong signal strength.
  // The range is capped between 0 and 4 to ensure that a change in the value
  // indicates a non-negligible change in the signal quality.
  optional int32 signal_strength = 3;
}
